src/machine_charm.py

```diff
@@ -88,11 +88,27 @@ def __post_init__(self):
 
         self.entries = entries
 
-    def entry(self, owner: str, endpoint_name: str) -> Optional[_SnapFstabEntry]:
+    def entry(self, owner: str, endpoint_name: Optional[str]) -> Optional[_SnapFstabEntry]:
         """Find and return a specific entry if it exists."""
-        entries = [
-            e for e in self.entries if e.owner == owner and endpoint_name in e.endpoint_source
-        ]
+        entries = [e for e in self.entries if e.owner == owner]
+
+        if len(entries) > 1 and endpoint_name:
+            # If there's more than one entry, the endpoint name may not directlly map to
+            # the sourcce *or* path. charmed-kafka uses 'logs' as the plug name, and maps
+            # .../common/logs to .../log inside Grafana Agent
+            #
+            # In this case, for a cheap comparison (rather than implementing some recursive
+            # LCS just for this, convert all possible endpoint sources into a list of unique
+            # characters, as well as the endpoint name, and build a dict of entries with
+            # a value that's the length of the intersection, the pick the first one.
```

Comment:
Could you add a "why" to the docstring? I don't understand why we're doing this.

Reply:
Done.